lib/auth/auth_with_roles.go

```diff
@@ -2669,6 +2669,11 @@ func (a *ServerWithRoles) SetAccessRequestState(ctx context.Context, params type
 	if err := a.action(apidefaults.Namespace, types.KindAccessRequest, types.VerbUpdate); err != nil {
 		return trace.Wrap(err)
 	}
+
+	if params.State.IsPromoted() {
```

Comment:
`SetAccessRequestState` changes the state to the provided one, `SubmitAccessReview` creates a review and proposes a new state that is applied to the request if all conditions are met.

Reply:
`SetAccessRequestState` used to be the only way to approve access requests.  Originally, approving access requests was a resource-level permission that let you approve _any_ access request unilaterally.  We added the review system later on, with its custom review permissions and thresholds so that individuals could be granted limited review permissions to a subset of requests.

You can think of `SetAccessRequestState` as the "admin" version of approval.  Also, some of our users have automated systems they've written in-house that automatically approve requests via this method.